### PR TITLE
Build: support DeePMD-kit C library

### PR DESCRIPTION
**Pull request by njzjz**
See https://github.com/deepmodeling/deepmd-kit/discussions/2124 for technical details.

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -36,8 +36,13 @@ jobs:
...
+      - name: Download DeePMD-kit
+        run: |
+          mkdir -p ${{ runner.temp }}
+          wget https://github.com/deepmodeling/deepmd-kit/releases/download/v2.2.0.b0/libdeepmd_c.tar.gz -O ${{ runner.temp }}/libdeepmd_c.tar.gz --no-check-certificate
+          tar vxzf ${{ runner.temp }}/libdeepmd_c.tar.gz -C ${{ runner.temp }}
       - name: Build
         run: |
           git config --global --add safe.directory `pwd`
-          cmake -B build -DBUILD_TESTING=ON ${{ matrix.build_args }}
+          cmake -B build -DBUILD_TESTING=ON ${{ matrix.build_args }} -DDeePMD_DIR=${{ runner.temp }}/libdeepmd_c
```
Comment (caic99) [inline .github/workflows/build_test.yml]: There is no deepmd building test yet. Maybe you should add a test first.

Comment (njzjz) [reply .github/workflows/build_test.yml]: I don't quite understand the current implementation. Where is `DeepPot.init` called?

Comment (njzjz) [reply .github/workflows/build_test.yml]: I only see `DeepPot.compute` is called below:

https://github.com/deepmodeling/abacus-develop/blob/af20f70cccfecfe3bcf5211eae5d012b4cd37c7e/source/module_esolver/esolver_dp.cpp#L50

Comment (caic99) [reply .github/workflows/build_test.yml]: > I don't quite understand the current implementation. Where is `DeepPot.init` called?

Hi @YuLiu98, would you take a look on this?

Comment (YuLiu98) [reply .github/workflows/build_test.yml]: > > I don't quite understand the current implementation. Where is `DeepPot.init` called?
> 
> Hi @YuLiu98, would you take a look on this?

`DeepPot.init` is called here: 
https://github.com/deepmodeling/abacus-develop/blob/af20f70cccfecfe3bcf5211eae5d012b4cd37c7e/source/module_esolver/esolver_dp.h#L16

Comment (YuLiu98) [reply .github/workflows/build_test.yml]: Hi @njzjz @caic99, is this PR complete? Or still missing the corresponding test?

Comment (caic99) [reply .github/workflows/build_test.yml]: > Hi @njzjz @caic99, is this PR complete? Or still missing the corresponding test?

@YuLiu98 Yes, currently there is no building test for deepmd add-on. Maybe you can submit a new PR on it, and please check with Prof Chen first. @mohanchen 

@njzjz I suggest not introducing building tests in this PR, for the implementation does not conform to the current test system. What do you think?

Comment (njzjz) [reply .github/workflows/build_test.yml]: I feel fine to revert it.

Comment (njzjz) [reply .github/workflows/build_test.yml]: Done.

**Comment by caic99** [comment]
Is C-style interface required by default? @YuLiu98 plz review

**Comment by njzjz** [comment]
The new interface is used by default. It will fall back to the old interface if the new interface is not found.

**Review by YuLiu98 — APPROVED** [review]

**Review by dyzheng — APPROVED** [review]

**Pull request merged**